programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt

```diff
@@ -0,0 +1,145 @@
+package twilio.flutter.twilio_programmable_video
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.ImageDecoder
+import android.graphics.ImageFormat
+import android.graphics.Matrix
+import android.graphics.Rect
+import android.graphics.YuvImage
+import android.os.Build
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+import tvi.webrtc.VideoFrame
+import tvi.webrtc.YuvConverter
+
+/**
+ * Converts a [tvi.webrtc.VideoFrame] to a Bitmap. This method must be called from a thread with a
+ * valid EGL context when the frame buffer is a [VideoFrame.TextureBuffer].
+ */
+fun VideoFrame.toBitmap(): Bitmap? {
+    val i420Buffer = if (buffer is VideoFrame.TextureBuffer) {
+        val yuvConverter = YuvConverter()
+        val buffer = yuvConverter.convert(buffer as VideoFrame.TextureBuffer)
+        yuvConverter.release()
+        buffer
+    } else {
+        buffer.toI420()
+    }
+    val yuvImage = i420ToYuvImage(
+            i420Buffer,
+            buffer.width,
+            buffer.height
+    )
+    val stream = ByteArrayOutputStream()
+    val rect =
+            Rect(0, 0, yuvImage.width, yuvImage.height)
+
+    // Compress YuvImage to jpeg
+    yuvImage.compressToJpeg(rect, 100, stream)
+
+    // Convert jpeg to Bitmap
+    val imageBytes = stream.toByteArray()
+    var bitmap = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+        val buffer = ByteBuffer.wrap(imageBytes)
+        val src =
+                ImageDecoder.createSource(buffer)
+        try {
+            ImageDecoder.decodeBitmap(src)
+        } catch (e: IOException) {
+            e.printStackTrace()
+            return null
+        }
+    } else {
+        BitmapFactory.decodeByteArray(imageBytes, 0, imageBytes.size)
+    }
+    val matrix = Matrix()
+
+    // Apply any needed rotation
+    matrix.postRotate(rotation.toFloat())
+    bitmap = Bitmap.createBitmap(
+            bitmap!!, 0, 0, bitmap.width, bitmap.height, matrix, true
+    )
+    return bitmap
+}
+
+fun VideoFrame.toJpeg(): ByteArray? {
+    val bitmap = toBitmap()
+    return if (bitmap != null) {
+        val stream = ByteArrayOutputStream()
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, stream)
```

Comment:
Same as above

Reply:
Actually, its roughly 740 kb for iOS and about 500 kb for Android.